core/src/test/java/org/elasticsearch/action/IndicesRequestIT.java

```diff
@@ -248,7 +249,7 @@ public void testUpdateUpsert() {
         String indexOrAlias = randomIndexOrAlias();
         UpdateRequest updateRequest = new UpdateRequest(indexOrAlias, "type", "id").upsert("field", "value").doc("field1", "value1");
         UpdateResponse updateResponse = internalCluster().coordOnlyNodeClient().update(updateRequest).actionGet();
-        assertThat(updateResponse.isCreated(), equalTo(true));
+        assertThat(updateResponse.getOperation(), equalTo(DocWriteResponse.Operation.CREATE));
```

Comment:
Personally I'd do this with `assertEquals` as well but you can absolutely keep this if you prefer it. We certainly don't have a standard for when hamcrest is a good idea. I expect there are folks in the Elasticsearch project who prefer hamcrest all the time so they can have consistent argument ordering. I don't, only using it for more complex assertions. But, yeah, if you like hamcrest then keep it this way.